### DISCUSSION TITLE
Adding an external spring to an MD simulation

### DISCUSSION THREAD

**Discussion by amruthesht** [glotzerlab/hoomd-blue#1744] [Support] [answered]
Hi all,

I was trying to attach a few particles (or center of masses of molecules) in my simulation box to a stiff spring acting as an external force on my system. I see that such functionality is available through `hoomd.hpmc.external.field.Harmonic` for the `hpmc` module. I was wondering if anything similar is available for the `md` module or if there is a workaround with existing classes and functions.

Any help will be greatly appreciated. Thank you!

Best,
Amru

**Comment by tcmoore3** [comment, marked as answer]
Hi Amru,

The way to do this in MD is to add additional particles to your system to act as restraint sites and add the appropriate bonding potentials between them and your particles to restrain. See [this how-to page](https://hoomd-blue.readthedocs.io/en/latest/howto/compute-the-free-energy-of-solids.html) for the general recipe.

Best,
Tim

**Reply by amruthesht** [reply]
Thank you very much, that was very helpful and worked out well!

**Comment by tcmoore3** [comment]
Hi Amru,

The way to do this in MD is to add additional particles to your system to
act as restraint sites and add the appropriate bonding potentials between
them and your particles to restrain. See [this how-to page](
https://hoomd-blue.readthedocs.io/en/latest/howto/compute-the-free-energy-of-solids.html)
for the general recipe.

Best,
Tim
